.github/workflows/smack-ci.yaml

```diff
@@ -48,8 +48,74 @@ jobs:
         run: |
           ./format/run-clang-format.py -r lib/smack include/smack tools share/smack/include share/smack/lib test examples
           flake8 test/regtest.py share/smack/ --extend-exclude share/smack/svcomp/,share/smack/reach.py
-      
+
       - name: compile and test
         env:
           TRAVIS_ENV: ${{ matrix.travis_env }}
         run: ./bin/build.sh
+
+  # Shaobo: I had to create a per-branch job because matrix doesn't work. The
+  # reason is that `if` expression is evaluated before matrix is expanded.
+  # See:
+  # https://github.community/t/matrix-cannot-be-used-in-jobs-level-if/17177/2
```

Comment:
Does this really justify the duplication?

I don’t see how matrix would have been the best solution anyways. This script will be run on each branch as is, since each branch is a separate workflow.

It seems like all you need here is to introduce a variable to hold the right Docker image tag, and to read that variable in the `docker_build` step. This variable would be set to `stable` or `latest`, depending on whether the branch is `master` or `develop`. Github actions seems capable of defining variables which depend on branch name, e.g., combining `set-env` or `set-output` with some shell command that generates the tag based on branch name.

https://stackoverflow.com/questions/58033366/how-to-get-current-branch-within-github-actions/58035262

https://www.techiediaries.com/get-current-branch-name-github-actions-github_ref-bash/

Reply:
Thank you for the suggestion. I've pushed an implementation based on `set-output` (`set-env` is deprecated.)